commit 105c7ca90d4152942e0798580a37f736d02faa22
Author: Jia Tan <jiat0218@gmail.com>
Date:   Thu Oct 19 16:23:32 2023 +0800

    Build: Remove check for COND_CHECK_CRC32 in check/Makefile.inc.
    
    Currently crc32 is always enabled, so COND_CHECK_CRC32 must always be
    set. Because of this, it makes the recent change to conditionally
    compile check/crc_clmul.c appear wrong since that file has CLMUL
    implementations for both CRC32 and CRC64.

diff --git a/src/liblzma/check/Makefile.inc b/src/liblzma/check/Makefile.inc
index 7780d20d..6186e10a 100644
--- a/src/liblzma/check/Makefile.inc
+++ b/src/liblzma/check/Makefile.inc
@@ -1,56 +1,56 @@
 ##
 ## Author: Lasse Collin
 ##
 ## This file has been put into the public domain.
 ## You can do whatever you want with this file.
 ##
+## Note: There is no check for COND_CHECK_CRC32 because
+## currently crc32 is always enabled.
 
 EXTRA_DIST += \
 	check/crc32_tablegen.c \
 	check/crc64_tablegen.c
 
 liblzma_la_SOURCES += \
 	check/check.c \
 	check/check.h \
 	check/crc_common.h
 
-if COND_CHECK_CRC32
 if COND_SMALL
 liblzma_la_SOURCES += check/crc32_small.c
 else
 liblzma_la_SOURCES += \
 	check/crc32_table.c \
 	check/crc32_table_le.h \
 	check/crc32_table_be.h
 if COND_ASM_X86
 liblzma_la_SOURCES += check/crc32_x86.S
 else
 liblzma_la_SOURCES += check/crc32_fast.c
 if COND_CRC_CLMUL
 liblzma_la_SOURCES += check/crc_clmul.c
 endif
 endif
 endif
-endif
 
 if COND_CHECK_CRC64
 if COND_SMALL
 liblzma_la_SOURCES += check/crc64_small.c
 else
 liblzma_la_SOURCES += \
 	check/crc64_table.c \
 	check/crc64_table_le.h \
 	check/crc64_table_be.h
 if COND_ASM_X86
 liblzma_la_SOURCES += check/crc64_x86.S
 else
 liblzma_la_SOURCES += check/crc64_fast.c
 endif
 endif
 endif
 
 if COND_CHECK_SHA256
 if COND_INTERNAL_SHA256
 liblzma_la_SOURCES += check/sha256.c
 endif
 endif